src/ui_basic/checkbox.cc

```diff
@@ -84,16 +83,15 @@ void Statebox::layout() {
 		}
 		rendered_text_ =
 		   label_text_.empty() ?
-		      nullptr :
-		      UI::g_fh->render(as_richtext_paragraph(label_text_, UI::FontStyle::kLabel),
+            nullptr :
+            UI::g_fh->render(as_richtext_paragraph(label_text_, UI::FontStyle::kLabel),
 		                       text_width(get_w(), pic_width));
 		if (rendered_text_) {
 			w = std::max(rendered_text_->width() + kPadding + pic_width, w);
 			h = std::max(rendered_text_->height(), h);
 		}
-		rendered_width_ = w;
-		set_desired_size(rendered_width_, h);
-		set_size(rendered_width_, h);
+		set_desired_size(w, h);
+		set_size(w, h);
```

Comment:
Remove the call to `set_size(w, h)`, which is eventually called during `set_desired_size(w, h)`. This fixes the options bug in a much simpler way.